registry/zookeeper/service_discovery.go

```diff
@@ -64,17 +64,18 @@ type zookeeperServiceDiscovery struct {
 
 // newZookeeperServiceDiscovery the constructor of newZookeeperServiceDiscovery
 func newZookeeperServiceDiscovery(url *common.URL) (registry.ServiceDiscovery, error) {
+	rp := url.GetParam(constant.ZookeeperRootPath, rootPath)
```

Comment:
This param should be injected by **config/ReferenceConfig.Refer** and **config/ServiceConfig.Export**.

Reply:
I have looked at the source code of Dubbo Java, and rootPath is only related to the registryCenter ZookeeperServiceDiscovery

RootPath is only used during ZookeeperServiceDiscovery initialization, and providers and consumers do not care about this param

Reply:
Which configuration style do you prefer to specify **ZookeeperRootPath** if you do not let providers and consumers inject this param?

Reply:
Sorry, I added a constant based on **ZookeeperKey**. What's the problem with this？
It is transparently transmitted to zookeeperServiceDiscovery through **RegistryConfig.Params**

Reply:
My bad, I did not describe it well. I thought you were going to add a generic feature. Using **RegistryConfig.Params** could work. **ZookeeperRootPath** should be "zookeeper.rootPath" like keys related to nacos(https://github.com/apache/dubbo-go/blob/5ac057d3ca6c3c0ea1fa355c6dcdf64a0813c6b9/common/constant/key.go#L246). Maybe we should add comments on **RegistryConfig.Params** since it is not intuitive to other users if there is no document explaining it.

Reply:
ok, **ZookeeperRootPath** has been modified to "zookeeper. rootPath",  and **RegistryConfig.Params** also added comments .  
Help review again ~

Reply:
`RegistryConfig.group` is also designed for users to specify the root path. 

So I think `ZookeeperRootPath = "zookeeper.rootPath"` is not necessary, directly using `group` would be fine. Or at least we need to check `group` if 'rootPath' is not set.

Reply:
The java implementation checked both rootPath and group. Maybe for some legacy consideration, for dubbogo, I personally think simplify check group is enough for RegistryConfig.group is designed for this scenario.

```java
 public static String getRootPath(URL registryURL) {
        String group = ROOT_PATH.getParameterValue(registryURL);
        if (group.equalsIgnoreCase(DEFAULT_GROUP)) {
            group = GROUP_PATH.getParameterValue(registryURL);
            if (!group.startsWith(PATH_SEPARATOR)) {
                group = PATH_SEPARATOR + group;
            }
        }
        return group;
    }
```

Reply:
I see that currently zookeeper service_discovery. go does not use **group**, thinking about the principle of minimum change, without considering group.
Do I need to add the **group** parameter logic as well? 
Like Dubbo Java:
```Java
public static String getRootPath(URL registryURL) {
    String group = ROOT_PATH.getParameterValue(registryURL);
    if (group.equalsIgnoreCase(DEFAULT_GROUP)) {
        group = GROUP_PATH.getParameterValue(registryURL);
        if (!group.startsWith(PATH_SEPARATOR)) {
            group = PATH_SEPARATOR + group;
        }
    }
    return group;
}
```

Reply:
replace zk **rootPath** with  **RegistryConfig.group**
Please review again